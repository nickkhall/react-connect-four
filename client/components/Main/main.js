import React from 'react';

const Main = ({
  children,
  classes
}) => (
  <main className={classes.main}>
    <div className={classes.toolbar} />
    { children }
  </main>
);

export default Main;
