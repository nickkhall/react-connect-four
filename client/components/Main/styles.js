export default theme => ({
  main: {
    flex: 1,
    display: 'flex',
    flexDirection: 'column',
    maxWidth: '100%'
  },
  toolbar: theme.mixins.toolbar
});
