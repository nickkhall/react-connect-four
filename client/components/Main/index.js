// React Router
import { withRouter } from 'react-router-dom';

// MUI Components
import { withStyles } from '@material-ui/core/styles';

// Components
import Main from './main';

// Local styles
import styles from './styles';

export default withRouter(
  withStyles(styles)(Main)
);
