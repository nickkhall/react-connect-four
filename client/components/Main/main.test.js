import React from 'react';

// Test utils
import {
  testCommonComponentAttrs,
  getComponentWrapper
} from 'utils/tests';

// Components
import Main from './main';

const mockNotificationID = 'foo';
const mockNotification = {
  id: mockNotificationID,
  key: mockNotificationID
};

const defaultProps = {
  classes: {},
  children: [
    <p key="One">One</p>,
    <p key="Two">Two</p>
  ],
  currentNotification: mockNotificationID,
  notifications: [mockNotification],
  queueRouteChange: jest.fn()
};

let wrapper;

describe('<Main /> component', () => {
  testCommonComponentAttrs(Main, defaultProps);

  beforeEach(() => {
    wrapper = getComponentWrapper(Main, defaultProps);
  });

  describe('rendering', () => {
    it('should render children', () => {
      expect(wrapper.find('p')).toHaveLength(2);
    });
  });
});
