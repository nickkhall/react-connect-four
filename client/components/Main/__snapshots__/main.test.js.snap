// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> component when rendering should render a component matching the snapshot 1`] = `
<main>
  <div />
  <p
    key="One"
  >
    One
  </p>
  <p
    key="Two"
  >
    Two
  </p>
</main>
`;
